relay-sampling/src/lib.rs

```diff
@@ -84,1522 +84,39 @@ pub use dsc::DynamicSamplingContext;
-    }
-
-    #[test]
-    /// test that the correct match is performed when replay id is present in the dsc.
-    fn test_sampling_match_with_trace_replay_id() {
-        let event = mocked_event(EventType::Transaction, "healthcheck", "1.1.1", "testing");
-        let dsc = mocked_dynamic_sampling_context(
-            "root_transaction",
-            "1.1.1",
-            "prod",
-            "vip",
-            "user-id",
-            Some(Uuid::new_v4()),
-        );
-
-        let result = match_against_rules(
-            &mocked_sampling_config_with_rules(vec![SamplingRule {
-                condition: and(vec![not(eq_null("trace.replay_id"))]),
-                sampling_value: SamplingValue::SampleRate { value: 1.0 },
-                ty: RuleType::Trace,
-                id: RuleId(1),
-                time_range: Default::default(),
-                decaying_fn: Default::default(),
-            }]),
-            &event,
-            &dsc,
-            Utc::now(),
-        );
-        assert_trace_match!(result, 1.0, dsc, 1)
-    }
-
-    #[test]
-    /// Tests that no match is done when there are no matching rules.
-    fn test_get_sampling_match_result_with_no_match() {
-        let sampling_config = mocked_sampling_config(SamplingMode::Received);
-        let event = mocked_event(EventType::Transaction, "transaction", "2.0", "");
-
-        let result = merge_configs_and_match(
-            true,
-            Some(&sampling_config),
-            None,
-            None,
-            Some(&event),
-            Utc::now(),
-        );
-        assert_no_match!(result);
-    }
-
-    #[test]
-    /// Tests that matching is still done on the transaction rules in case trace params are invalid.
-    fn test_get_sampling_match_result_with_invalid_trace_params() {
-        let sampling_config = mocked_sampling_config(SamplingMode::Received);
-        let root_project_sampling_config =
-            mocked_root_project_sampling_config(SamplingMode::Received);
-        let dsc = mocked_simple_dynamic_sampling_context(Some(1.0), Some("1.0"), None, None);
-
-        let event = mocked_event(EventType::Transaction, "foo", "2.0", "");
-        let result = merge_configs_and_match(
-            true,
-            Some(&sampling_config),
-            Some(&root_project_sampling_config),
-            None,
-            Some(&event),
-            Utc::now(),
-        );
-        assert_transaction_match!(result, 0.5, event, 3);
-
-        let event = mocked_event(EventType::Transaction, "healthcheck", "2.0", "");
-        let result = merge_configs_and_match(
-            true,
-            Some(&sampling_config),
-            None,
-            Some(&dsc),
-            Some(&event),
-            Utc::now(),
-        );
-        assert_transaction_match!(result, 0.1, event, 1);
-    }
-
-    #[test]
-    /// Tests that a match with early return is done in the project sampling config.
-    fn test_get_sampling_match_result_with_project_config_match() {
```

Comment:
removed tests using configs as input, as we now serve rules directly and I have tests for that.